components/ide/jetbrains/gateway-plugin/src/main/kotlin/io/gitpod/jetbrains/gateway/GitpodConnectionProvider.kt

```diff
@@ -296,7 +296,11 @@ class GitpodConnectionProvider : GatewayConnectionProvider {
                                     )
                                     clientHandle.clientClosed.advise(connectionLifetime) {
                                         application.invokeLater {
-                                            connectionLifetime.terminate()
+                                            GlobalScope.launch {
```

Comment:
is it valid usage of `GlobalScope.launch`, I remember it should be used for application wide coroutines, there were some gotchas with it